File: hangman/python/gameLoop.py
from wordToGuess import WordToGuess
from hungDude import HungDude

class gameLoop:

	def __init__(self):
		self.word = self.getWordToGuess()
		self.hd = HungDude()
	
	def mainLoop(self):
		word=self.word
		while not word.wordIsComplete() and word.guessesLeft():
			print 'The word so far is \'%s\'.'%word.getGuessedWord()
			guess = self.getInput()
			con = word.guessLetter(guess)
			self.printHD()
			if con:
				print '\'%s\' is in the word.'%guess
			else:
				print '\'%s\' is not in the word.'%guess
				print 'You have %d guesses left.'%word.guessesLeft()
			print ''
		if word.wordIsComplete():
			print 'Congratulations, you win!'
			return True
		print 'Sorry, you lose!'
		print 'The word was \'%s\'.'%word.getWord()
		return False
	
	def printHD(self):
		for x in self.hd.getFull(self.word.getWrongGuesses()):
			print x
	def getInput(self):
		ret = False
		while not ret:
			guess = raw_input('Please enter a letter: ')
			ret = self.checkInput(guess)
			if not ret:
				print 'That guess is invalid.'
			else:
				guess = guess.lower()
				if self.word.checkForLetterInWrongGuesses(guess):
					ret = False
					print 'You have already guessed \'%s\''%guess
		return guess
	
	def checkInput(self, guess):
		isGood = True
		isGood = isGood and len(guess) == 1
		isGood = isGood and str(guess) == guess
		isGood = isGood and guess.isalpha()
		return isGood
	def getWordToGuess(self):
		return WordToGuess()


if __name__ == '__main__':
	gl=gameLoop()
	gl.mainLoop()

File: hangman/cpp/GameLoop.cpp
#include "WordToGuess.cpp"
#include <cstdlib>
using namespace std;

class GameLoop{
	public: //temporary
	WordToGuess word;

	public:
		GameLoop();
		void mainLoop();
		char getInput();
		bool checkInput(char guess);
};

GameLoop::GameLoop(){
}

void GameLoop::mainLoop(){
	while(true){
		cout << "The word so far is " << word.getGuessedWord() << endl;
		char guess = getInput();
		bool con = word.guessLetter(guess);
		if(con){
			cout << guess << " is in the word." << endl;
		} else{
			cout << guess << " is not in the word." << endl;
			cout << "You have " << word.guessesLeft() << " guesses left." << endl;
		}
		cout << endl;
		if(word.wordIsComplete() || word.guessesLeft()<=0){
			break;
		}
	}
	if(word.wordIsComplete()){
		cout << "Congratulations, you win!" << endl;
	} else{
		cout << "Sorry, you lose!" << endl;
		cout << "The word was '" << word.getWord() << "'." << endl;
	}
}
char GameLoop::getInput(){
	char guess;
	while (true) {
		cout << "Please enter a letter: ";
		if (!(cin >> guess)) {
			cerr << "input error\n";
			return 1;
		}
		bool ret = checkInput(guess);
		if(!ret){
			cout << "That guess is invalid." << endl;
		} else if(word.checkForLetterInGuesses(guess)){
			ret = false;
			cout << "You have already guessed '" << guess << "'." << endl;
		}
		if(ret){break;}
	}
	return guess;
}
bool GameLoop::checkInput(char guess){
	bool isGood = true;
	return isGood;
}

int main(){
	GameLoop gl;
	gl.mainLoop();
	return 0;
}
